File: core/baker-interface/src/main/scala/com/ing/baker/runtime/defaultinteractions/TimerInteraction.scala
package com.ing.baker.runtime.defaultinteractions

import com.ing.baker.runtime.catseffect.AsyncSupport
import scala.concurrent.duration.FiniteDuration

class TimerInteraction[F[_]](skipWait: Boolean)(implicit F: AsyncSupport[F]) {

  class TimeWaited

  def apply(WaitTime: FiniteDuration): F[TimeWaited] = {
    if(skipWait)
      F.pure(new TimeWaited)
    else
      F.map(F.sleep(WaitTime))(_ => new TimeWaited)
  }
}

File: core/baker-recipe-dsl/src/main/scala/com/ing/baker/recipe/javadsl/interactions/TimerInteraction.java
package com.ing.baker.recipe.javadsl.interactions;

        import com.ing.baker.recipe.annotations.FiresEvent;
        import com.ing.baker.recipe.annotations.RequiresIngredient;
        import com.ing.baker.recipe.javadsl.Interaction;

        import java.time.Duration;

public interface TimerInteraction extends Interaction {
    @FiresEvent(oneOf = TimeWaited.class)
    public TimeWaited apply(@RequiresIngredient("WaitTime") Duration WaitTime);
}